### PR TITLE
Support block machines with block size 1

### PR DESCRIPTION
**Pull request by Schaeff**
- allow 1 as the block machine latch
- add test

### CODE REVIEW

File: executor/src/witgen/machines/block_machine.rs [hunk 1]
```diff
@@ -50,38 +50,35 @@ impl<T: FieldElement> BlockMachine<T> {
         global_range_constraints: &BTreeMap<&PolynomialReference, RangeConstraint<T>>,
     ) -> Option<Box<Self>> {
         for id in connecting_identities {
-            if let Some(sel) = &id.right.selector {
```
Comment (Schaeff) [inline executor/src/witgen/machines/block_machine.rs:53]: diff looks big but I just removed this condition

File: executor/src/witgen/machines/block_machine.rs [hunk 2]
```diff
@@ -50,38 +50,35 @@ impl<T: FieldElement> BlockMachine<T> {
...
-                        processing_sequence_cache: ProcessingSequenceCache::new(
-                            period,
-                            identities.len(),
-                        ),
-                    };
-                    // Append a block so that we do not have to deal with wrap-around
-                    // when storing machine witness data.
-                    machine.append_new_block(fixed_data.degree).unwrap();
+            // TODO we should check that the other constraints/fixed columns are also periodic.
+            if let Some(period) = try_to_boolean_periodic_selector(&id.right.selector, fixed_data) {
```
Comment (chriseth) [inline executor/src/witgen/machines/block_machine.rs:54]: But now the function only returns the period and not the selector. Maybe rename?

**Comment by chriseth** [comment]
Looks good, only maybe rename, but also fine as is.

**Review by chriseth — APPROVED** [review]

File: executor/src/witgen/machines/block_machine.rs (hunk 2 above)
Comment (Schaeff) [reply executor/src/witgen/machines/block_machine.rs]: Done!

**Pull request merged**